### PR TITLE
fix(input, input-number): correctly handle '-' and '.' values when sanitizing number

### PR DESCRIPTION
**Pull request by benelan**
**Related Issue:** #6270

## Summary
Fixes issues in input and input-number involving how to handle `-` and `.` values. We don't want to set it as the `value` property because they aren't numbers, but we also don't want to sanitize them because people should be allowed to type negative and decimal numbers. Added a couple code comments for clarity

### CODE REVIEW

File: src/components/input-number/input-number.tsx [hunk 1]
```diff
@@ -806,6 +806,7 @@ export class InputNumber
     };
 
     const sanitizedValue = sanitizeNumberString(
+        // no need to delocalize a string that ia already in latn numerals
```
Comment (eriklharper) [inline src/components/input-number/input-number.tsx:809]: typo: "is" instead of "ia"

**Review by eriklharper — APPROVED** [review]
Looks good, tested locally.  Just noticed a typo in the comment, other than that, ready to merge :)

File: src/components/input-number/input-number.tsx (hunk 1 above)
Comment (benelan) [reply src/components/input-number/input-number.tsx:809]: Nice catch on the typo. Its breaking some tests which I'm looking into before merging

File: src/components/input-number/input-number.e2e.ts [hunk 2]
```diff
@@ -914,13 +914,16 @@ describe("calcite-input-number", () => {
...
 
-  it("allows clearing value with an empty string", async () => {
+  it("allows negative numbers after clearing value with an empty string", async () => {
     const page = await newE2EPage();
     await page.setContent(html`<calcite-input-number value="1"></calcite-input-number>`);
     const input = await page.find("calcite-input-number");
     input.setProperty("value", "");
     await page.waitForChanges();
     expect(await input.getProperty("value")).toBe("");
+    typeNumberValue(page, "-123");
```
Comment (jcfranco) [inline src/components/input-number/input-number.e2e.ts:924]: This needs to be awaited on, right?

File: src/components/input/input.e2e.ts [hunk 3]
```diff
@@ -1070,13 +1070,16 @@ describe("calcite-input", () => {
...
 
-    it("allows clearing value with an empty string", async () => {
+    it("allows negative numbers after clearing value with an empty string", async () => {
       const page = await newE2EPage();
       await page.setContent(html`<calcite-input type="number" value="1"></calcite-input>`);
       const input = await page.find("calcite-input");
       input.setProperty("value", "");
       await page.waitForChanges();
       expect(await input.getProperty("value")).toBe("");
+      typeNumberValue(page, "-123");
```
Comment (jcfranco) [inline src/components/input/input.e2e.ts:1080]: Missing await here too.

**Pull request merged**